python/src/multi_agent_orchestrator/agents/supervisor_agent.py

```diff
@@ -253,7 +253,7 @@ def _format_agents_memory(self, agents_history: list[ConversationMessage]) -> st
         return ''.join(
             f"{user_msg.role}:{user_msg.content[0].get('text','')}\n"
             f"{asst_msg.role}:{asst_msg.content[0].get('text','')}\n"
-            for user_msg, asst_msg in zip(agents_history[::2], agents_history[1::2])
+            for user_msg, asst_msg in zip(agents_history[::2], agents_history[1::2], strict=True)
```

Comment:
Adding strict parameter to make sure `zip` function will unpack expected values OR will raise `ValueError` if not.